src/components/CreatePostForm.tsx

```diff
@@ -0,0 +1,65 @@
+// Render Prop
+import React from 'react';
+import { Formik, Form, Field, ErrorMessage } from 'formik';
+import { PostItemInterface } from '../interfaces/Post';
+
+
+const CreatePostForm = ({
+  onSavePost
+}: { onSavePost: (i: PostItemInterface) => void}) => (
+  <div>
+    <h1>New Post</h1>
+    <Formik
+      initialValues={{ title: '', body: '', id: '' }}
+      validate={(values: PostItemInterface) => {
+        const errors: PostItemInterface = {};
+        if (!values.title) {
+          errors.title = 'Required';
+        } else if (!values.body) {
+          errors.body = 'Required';
+        }
+        return errors;
+      }}
+      onSubmit={(values, { setSubmitting }) => {
+        onSavePost({
+          ...values
+        })
+        setTimeout(() => {
```

Comment:
While I appreciate the gesture of simulating a network response, why not using the API directly?

Reply:
i let this just to emulate the time the call to api could take